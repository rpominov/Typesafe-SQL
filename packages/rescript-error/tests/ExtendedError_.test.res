open Jest

test("Not_found->toJsError", () => {
  expect(Not_found->ExtendedError.toJsError)->toMatchSnapshot

  try {
    raise(Not_found)
  } catch {
  | exn => expect(exn->ExtendedError.toJsError)->toMatchSnapshot
  }
})

test("Invalid_argument->toJsError", () => {
  expect(Invalid_argument("test")->ExtendedError.toJsError)->toMatchSnapshot

  try {
    raise(Invalid_argument("test"))
  } catch {
  | exn => expect(exn->ExtendedError.toJsError)->toMatchSnapshot
  }
})

test("toJsError + stack (not raised)", () => {
  expect(
    Not_found
    ->ExtendedError.toJsError
    ->ExtendedError.stack
    ->Js.String2.split("\n")
    ->Js.Array2.unsafe_get(1)
    // When not raised, the exn won't have a Error property,
    // so the Error will be created inside toJsError(),
    // and the stack will point to the library file
    ->Js.String2.includes("ExtendedError.bs.js"),
  )->toBe(true)
})

test("toJsError + stack", () => {
  try {
    raise(Not_found)
  } catch {
  | exn =>
    expect(
      exn
      ->ExtendedError.toJsError
      ->ExtendedError.stack
      ->Js.String2.split("\n")
      ->Js.Array2.unsafe_get(1)
      // When exn is raised, stack points to the file it was raised from
      ->Js.String2.includes("ExtendedError_.test.bs.js"),
    )->toBe(true)
  }
})
