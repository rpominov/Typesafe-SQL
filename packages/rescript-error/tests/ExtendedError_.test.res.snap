// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom->toJsError 1`] = `
Object {
  "NAME": "ValidJsError",
  "VAL": [ReScript_Error: ExtendedError_.Custom/2(_1: {"test":123})],
}
`;

exports[`Custom->toJsError 2`] = `
Object {
  "NAME": "ValidJsError",
  "VAL": [ReScript_Error: ExtendedError_.Custom/2(_1: {"test":123})],
}
`;

exports[`Invalid_argument->toJsError 1`] = `
Object {
  "NAME": "ValidJsError",
  "VAL": [ReScript_Error: Invalid_argument(_1: "test")],
}
`;

exports[`Invalid_argument->toJsError 2`] = `
Object {
  "NAME": "ValidJsError",
  "VAL": [ReScript_Error: Invalid_argument(_1: "test")],
}
`;

exports[`Not_found->toJsError 1`] = `
Object {
  "NAME": "ValidJsError",
  "VAL": [ReScript_Error: Not_found],
}
`;

exports[`Not_found->toJsError 2`] = `
Object {
  "NAME": "ValidJsError",
  "VAL": [ReScript_Error: Not_found],
}
`;
